src/redis_lock/__init__.py

```diff
@@ -195,12 +213,22 @@ def extend(self, expire=None):
             New expiration time. If ``None`` - `expire` provided during
             lock initialization will be taken.
         """
-        if self._expire is None:
-            raise NotExpirable('The lock has no expiry time, so extending it '
-                               'makes no sense.')
-        if expire is None:
+        if expire is None and self._expire is not None:
             expire = self._expire
-        self._client.set(self._name, self._id, xx=True, ex=expire)
+        elif expire is None and self._expire is None:
```

Comment:
You can check for `expire is None` just once I think. It's one indentation level more but then it's clear what's the common part.